nautobot/extras/graphql/types.py

```diff
@@ -1,9 +1,31 @@
+import graphene
+from graphene_django import DjangoObjectType
 import graphene_django_optimizer as gql_optimizer
 
 from nautobot.extras.models import Status, Tag, DynamicGroup
 from nautobot.extras.filters import StatusFilterSet, TagFilterSet, DynamicGroupFilterSet
 
 
+class NautobotObjectType(DjangoObjectType):
+    url = graphene.String()
+
+    def resolve_url(self, args):
+        return self.get_absolute_url(api=True)
+
+    class Meta:
+        abstract = True
+
+
+class OptimizedNautobotObjectType(gql_optimizer.OptimizedDjangoObjectType):
+    url = graphene.String()
+
+    def resolve_url(self, args):
+        return self.get_absolute_url(api=True)
+
+    class Meta:
+        abstract = True
+
+
 class TagType(gql_optimizer.OptimizedDjangoObjectType):
```

Comment:
Should these types below be converted to use `OptimizedNautobotObjectType`?

Reply:
I'm not seeing any negative impact changing everything to `OptimizedNautobotObjectType`. I changed these and also updated `nautobot.core.graphql.generators.generate_schema_type()` to return an `OptimizedNautobotObjectType` subclass.